akka-persistence/src/main/scala/akka/persistence/journal/PersistencePluginProxy.scala

```diff
@@ -136,8 +131,57 @@ final class JournalProxy(config: Config) extends Actor with Stash with ActorLogg
       if (targetAtThisNode && address != selfAddress)
         becomeIdentifying(address)
     case InitTimeout ⇒
-    case msg ⇒
-      targetJournal.forward(msg)
+
+    case msg @ WriteMessages(messages, _, actorInstanceId) ⇒
+      implicit val ec = context.system.dispatcher
+      val sender = context.sender()
+      val f = ask(targetJournal, msg)(1.second)
+      f onComplete {
+        case Success(v) ⇒ sender ! v
+        case Failure(t) ⇒
+          sender ! WriteMessagesFailed(t)
+          messages foreach {
+            case a: AtomicWrite       ⇒ a.payload.foreach { p ⇒ sender ! WriteMessageFailure(p, t, actorInstanceId) }
+            case r: NonPersistentRepr ⇒ sender ! LoopMessageSuccess(r.payload, actorInstanceId)
+          }
+      }
+    case msg @ DeleteMessagesTo(_, toSeqNr, _) ⇒
+      implicit val ec = context.system.dispatcher
+      val sender = context.sender()
+      val f = ask(targetJournal, msg)(2.seconds)
+      f recover { case t ⇒ DeleteMessagesFailure(t, toSeqNr) } pipeTo sender
+    case msg @ ReplayMessages(_, _, _, _, _) ⇒
+      implicit val ec = context.system.dispatcher
+      val sender = context.sender()
+      val f = ask(targetJournal, msg)(2.seconds)
+      f recover { case t ⇒ ReplayMessagesFailure(t) } pipeTo sender
+
+    case msg @ LoadSnapshot(_, _, toSequenceNr) ⇒
+      implicit val ec = context.system.dispatcher
+      val sender = context.sender()
+      val f = ask(targetJournal, msg)(2.seconds)
+      f recover { case t ⇒ LoadSnapshotResult(None, toSequenceNr) } pipeTo sender
+    case msg @ SaveSnapshot(metadata, _) ⇒
+      implicit val ec = context.system.dispatcher
+      val sender = context.sender()
+      val f = ask(targetJournal, msg)(2.seconds)
+      f recover { case t ⇒ SaveSnapshotFailure(metadata, t) } pipeTo sender
+    case msg @ DeleteSnapshot(metadata) ⇒
+      implicit val ec = context.system.dispatcher
+      val sender = context.sender()
+      val f = ask(targetJournal, msg)(2.seconds)
+      f recover { case t ⇒ DeleteSnapshotFailure(metadata, t) } pipeTo sender
+    case msg @ DeleteSnapshots(_, criteria) ⇒
+      implicit val ec = context.system.dispatcher
+      val sender = context.sender()
+      val f = ask(targetJournal, msg)(2.seconds)
+      f recover { case t ⇒ DeleteSnapshotsFailure(criteria, t) } pipeTo sender
+
+    case msg ⇒ // is this necessary? what other messages can we get?
```

Comment:
for custom plugin defined messages, let's just forward them (we can't know if there will be a reply)